### ISSUE TITLE
Urgent: Home down because of invalid record (?)

### ISSUE THREAD

**Issue by icarito** [publiclab/plots2#2647]
```
Started GET "/" for 47.31.242.72 at 2018-04-21 19:46:30 +0000
Processing by HomeController#home as HTML
Read fragment views/feature_home-carousel (0.3ms)                                                                                                              Read fragment views/feature_home-banners (0.1ms)                                                                                                               Read fragment views/home-activity (0.1ms)                                                                                                                      Read fragment views/trending-tags (0.1ms)                                                                                                                        Rendered dashboard/_node_comment.html.erb (9.2ms)                                                                                                              Rendered dashboard/_node_meta.html.erb (3.2ms)                                                                                                                 Rendered dashboard/_node_wiki.html.erb (7.5ms)                                                                                                                 Rendered dashboard/_node_wiki.html.erb (2.0ms)                                                                                                                 Rendered dashboard/_activity.html.erb (22.7ms)
  Rendered home/home.html.erb within layouts/application (25.9ms)
Completed 500 Internal Server Error in 37ms (ActiveRecord: 3.4ms)

ActionView::Template::Error (undefined method `author' for nil:NilClass):
    3:                                                                                                                                                             4:     <% if node.is_a?(Node) %>
    5:       <div class="header-icon"><i class="fa fa-book"></i></div>
    6:       <p class="meta"><span><%= t('dashboard._node_wiki.new_page_by') %></span> <a href="/profile/<%= node.latest.author.name %>"><%= node.latest.author.name %></a></p>                                                                                                                                                   7:     <% else %>
    8:       <div class="header-icon"><i class="fa fa-pencil"></i></div>
    9:       <p class="meta"><%= t('dashboard._node_wiki.new_edit_by') %> <a href="/profile/<%= node.author.try(:name) %>"><%= node.author.try(:name) %></a></p>
  app/views/dashboard/_node_wiki.html.erb:6:in `_app_views_dashboard__node_wiki_html_erb___2292190031077852178_69979555127840'
  app/views/dashboard/_activity.html.erb:67:in `block in _app_views_dashboard__activity_html_erb__2360015329309354006_69979556424080'
  app/views/dashboard/_activity.html.erb:65:in `each'
  app/views/dashboard/_activity.html.erb:65:in `each_with_index'
  app/views/dashboard/_activity.html.erb:65:in `_app_views_dashboard__activity_html_erb__2360015329309354006_69979556424080'
  app/views/home/home.html.erb:118:in `block in _app_views_home_home_html_erb__1339280168507799139_35544200'
  app/views/home/home.html.erb:117:in `_app_views_home_home_html_erb__1339280168507799139_35544200'
  app/controllers/home_controller.rb:18:in `home'
```

### Please describe the problem (or idea)

> What happened just before the problem occurred? Or what problem could this idea solve?

> What did you expect to see that you didn't?

### Please show us where to look

https://publiclab.org/ ...

### What's your PublicLab.org username?

> This can help us diagnose the issue: 

### Browser, version, and operating system

> Many bugs are related to these -- please help us track it down and reproduce what you're seeing!

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site. 

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

**Comment by icarito** [comment]
I'm going to look for the invalid record with the rails console. Need to figure out what record I'm looking for.

**Comment by icarito** [comment]
Checking thru the latest nodes they seem to all have valid author fields... I'm going to try restarting the app.

**Comment by icarito** [comment]
These all look like valid users to me?
```
irb(main):004:0> Node.order(nid: :desc).limit(10).collect(&:author).collect(&:uid)
=> [519183, 531671, 531623, 531623, 530669, 1, 531523, 531503, 531494, 525300]
```

**Comment by icarito** [comment]
Yeah, valid users, then how are we getting nil for a node? :confused: 
```
irb(main):005:0> Node.order(nid: :desc).limit(10).collect(&:author).collect(&:name)                                                                                                                                                            
=> ["AeroceneFoundation", "myangels", "aashirana", "aashirana", "section80g", "warren", "kingsmenia", "shathishati", "peterparker2704", "stefannibrasil"]
```

**Comment by icarito** [comment]
`node.latest.author.name` seems to be the culprit `node` is `nil`

**Comment by pdurbin** [comment]
> ActionView::Template::Error (undefined method `author' for nil:NilClass):

My guess is that `node` is nil on this line:

https://github.com/publiclab/plots2/blob/fa16a69f8f475c101bc23325f4df7411c2172872/app/views/dashboard/_node_wiki.html.erb#L9

**Comment by Gauravano** [comment]
I think, I have reproduced the error

![error](https://user-images.githubusercontent.com/20878070/39089048-87eb91d6-45db-11e8-9fa7-5216ccfe8cb9.gif)

**Comment by jywarren** [comment]
Fixed, opening a PR to reproduce this -- maybe @Gauravano if you'd be interested in trying to write a function to disallow spamming of the final revision of a wiki page (instead alerting that an admin must delete the page) that would be amazing!!! THanks all for working on this.

**Comment by jywarren** [comment]
Awesome, just opened #2648.

**Comment by Gauravano** [comment]
Fixed in #2648
